### ISSUE TITLE
Sometimes throw java.lang.ClassCastException

### ISSUE THREAD

**Issue by sczyh30** [sczyh30/vertx-kue#1]
Sometimes the application sometimes would throw `java.lang.ClassCastException: java.lang.String cannot be cast to [Lio.vertx.redis.impl.Reply` like this(version: **3.3.0-SNAPSHOT**):

``` java
六月 11, 2016 7:47:55 下午 io.vertx.core.impl.ContextImpl
严重: Unhandled exception
java.lang.ClassCastException: java.lang.String cannot be cast to [Lio.vertx.redis.impl.Reply;
    at io.vertx.redis.impl.Reply.asType(Reply.java:124)
    at io.vertx.redis.impl.Reply.asType(Reply.java:156)
    at io.vertx.redis.impl.RedisConnection.handleReply(RedisConnection.java:376)
    at io.vertx.redis.impl.ReplyParser.handle(ReplyParser.java:180)
    at io.vertx.redis.impl.ReplyParser.handle(ReplyParser.java:21)
    at io.vertx.core.net.impl.NetSocketImpl.handleDataReceived(NetSocketImpl.java:312)
    at io.vertx.core.net.impl.VertxNetHandler.lambda$channelRead$24(VertxNetHandler.java:67)
    at io.vertx.core.impl.ContextImpl.lambda$wrapTask$3(ContextImpl.java:357)
    at io.vertx.core.impl.ContextImpl.executeFromIO(ContextImpl.java:234)
    at io.vertx.core.net.impl.VertxNetHandler.channelRead(VertxNetHandler.java:67)
    at io.vertx.core.net.impl.VertxNetHandler.channelRead(VertxNetHandler.java:31)
    at io.vertx.core.net.impl.VertxHandler.channelRead(VertxHandler.java:122)
    at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:349)
    at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:335)
    at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:327)
    at io.netty.channel.DefaultChannelPipeline$HeadContext.channelRead(DefaultChannelPipeline.java:1336)
    at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:349)
    at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:335)
    at io.netty.channel.DefaultChannelPipeline.fireChannelRead(DefaultChannelPipeline.java:911)
    at io.netty.channel.nio.AbstractNioByteChannel$NioByteUnsafe.read(AbstractNioByteChannel.java:123)
    at io.netty.channel.nio.NioEventLoop.processSelectedKey(NioEventLoop.java:544)
    at io.netty.channel.nio.NioEventLoop.processSelectedKeysOptimized(NioEventLoop.java:485)
    at io.netty.channel.nio.NioEventLoop.processSelectedKeys(NioEventLoop.java:399)
    at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:371)
    at io.netty.util.concurrent.SingleThreadEventExecutor$5.run(SingleThreadEventExecutor.java:742)
    at java.lang.Thread.run(Thread.java:745)
```

So far I haven't found out where the Exception throws. I'll try to repeat the exception and fix it.

**Comment by sczyh30** [comment]
Related to [vertx-redis-client#52](https://github.com/vert-x3/vertx-redis-client/issues/52).
Caused by: `hgetall` between a transaction lifecycle
The original bug seems not fixed absolutely in 3.3.0?

**Comment by sczyh30** [comment]
Caused by: invoke `hgetall` via `RedisClient` between a transaction

Solved
